test/cypress/support/commands.js

```diff
@@ -16,46 +16,98 @@ Cypress.Commands.add('getByDT', (element, otherSelector = '') => {
   return cy.get(`${otherSelector}[data-test="${element}"]`)
 })
 
-// NOTE: We can go a step further and not use UI to do repetitive tasks.
-// https://docs.cypress.io/guides/getting-started/testing-your-app.html#Fully-test-the-login-flow-%E2%80%93-but-only-once
-Cypress.Commands.add('giSignup', (userName, {
+function cyBypassUI (action, params) {
+  const query = Object.keys(params).reduce((query, param) => {
+    return query + `&${param}=${params[param]}`
+  }, `?action=${action}`)
+
+  cy.log(`Bypassing UI ::: ${action}`)
+
+  // Option 1 - wait for contracts to finish sync. - doesnt work.
+  // cy.getByDT('app').then(([el]) => {
+  //   cy.get(el).should('have.attr', 'data-sync', '')
+  // })
+  // cy.visit(`/app/bypass-ui?action=${action}${query}`)
+
+  // Option 2 - hardcore wait. it works, but it isn't the ideal solution.
+  // cy.wait(400); // eslint-disable-line
+  // cy.visit(`/app/bypass-ui?action=${action}${query}`)
+
+  // Option 3 - Navigate to /bypass-ui using Vue instead of cy.visit!
```

Comment:
@taoeffect  and @pieer please read this!

#### A quick recap of this PR joruney:

We have been trying to use `/bypass-ui` page to speed up our tests. The only issue is that cypress is so fast visiting `/bypass-ui` that sometimes information gets lost.

So we thought it was some problem with our code (contracts not fully synced)... but it's not! Why do I know that? Because after cypress fails to load a page data (ex: profile info) and I manually refresh it a second time, the information is there again! So this means, the contracts were indeed syncing. It has to be something else... maybe the way how `forageStorage` (indexedDB) works?

Well... so the idea to overcome this is explained on the code (Option 3) - using Vue to navigate to `/bypass-ui` without causing a page load/refresh.

So basically now when we run tests on cypress, a little button shows up on the corner, all the time. It's our "magic gate" to `/bypass-ui` page.

Please let me know if some further explanation about this is needed. I'll add it directly to this code.

![image](https://user-images.githubusercontent.com/14869087/73064685-b3761480-3e99-11ea-9c52-850bfeb19378.png)

Reply:
That's an interesting way of doing it. Avoiding the need for a `.visit` might also speed things up too, right? If it works, it works, I see no major issue of doing it this way at least.

Reply:
yes, it's slightly faster because we don't wait for the page to load all the stuff again.